yt/yt/tests/integration/queries/test_mock.py

```diff
@@ -202,6 +205,44 @@ def test_rows_limit(self, query_tracker):
         assert not q.get_result(1)["is_truncated"]
 
 
+class TestQueryTrackerBan(YTEnvSetup):
+    NUM_QUERY_TRACKER = 1
+    DELTA_DRIVER_CONFIG = {
+        "cluster_connection_dynamic_config_policy": "from_cluster_directory",
+    }
+    QUERY_TRACKER_DYNAMIC_CONFIG = {"state_check_period": 2000}
+
+    def _test_query_fails():
+        start_query("mock", "run_forever")
+        return False
+
+    @authors("mpereskokova")
+    def test_query_tracker_ban(self, query_tracker):
+        address = query_tracker.query_tracker.addresses[0]
+        set(f"//sys/query_tracker/instances/{address}/@banned", True)
+
+        with raises_yt_error() as err:
+            wait(TestQueryTrackerBan._test_query_fails)
+        assert err[0].contains_text("No alive peers found")
+        wait(lambda: get(f"//sys/query_tracker/instances/{address}/orchid/@banned"), ignore_exceptions=True)
+
+        guid = generate_uuid()
+        insert_rows("//sys/query_tracker/active_queries", [{
+            "query_id": guid,
+            "engine": "mock",
+            "query": "run_forever",
+            "incarnation": 0,
+            "state": "pending",
+            "settings": {}
+        }])
+        time.sleep(3)  # NB(mpereskokova): Sleep for multiple active_query_acquisition_periods, to ensure, that inserted query won't be launched
```

Comment:
Even though this is fine with me, we could expose the counter of query acquisition iterations into the orchid and wait for a required number of iterations to pass.